ghost/admin/app/helpers/history-event-fetcher.js

```diff
@@ -91,11 +91,30 @@ export default class HistoryEventFetcher extends Resource {
                 this.hasReachedEnd = true;
             }
 
-            actions.forEach((a) => {
+            let count = 1;
+
+            actions.reverse().forEach((a, index) => {
+                const nextAction = actions[index + 1] || null;
+
+                // depending on the similarity, add additional properties to be used on the frontend for grouping
+                // skip - used for hiding the event on the frontend
+                // count - the number of similar events which is added to the last item
+                if (nextAction || (!nextAction && actions[index - 1].skip)) {
+                    if (nextAction && a.resource_id === nextAction.resource_id && a.event === nextAction.event) {
+                        a.skip = true;
+                        count += 1;
+                    } else {
+                        if (count > 1) {
+                            a.count = count.toString();
+                            count = 1;
+                        }
+                    }
+                }
+
                 a.context = JSON.parse(a.context);
             });
 
-            this.data.push(...actions);
+            this.data.push(...actions.reverse());
```

Comment:
Should we also handle the "similar" actions that are right no the edge of the previously inserted actions? The situation I'm thinking about here is when the last action that we have previously loaded is similar to the very first action in the just fetched actions.